apiserver/facades/client/secretbackends/secrets.go

```diff
@@ -35,23 +39,42 @@ func (s *SecretBackendsAPI) checkCanAdmin() error {
 
 // AddSecretBackends adds new secret backends.
 func (s *SecretBackendsAPI) AddSecretBackends(args params.AddSecretBackendArgs) (params.ErrorResults, error) {
-	result := params.ErrorResults{}
+	result := params.ErrorResults{
+		Results: make([]params.ErrorResult, len(args.Args)),
+	}
 	if err := s.checkCanAdmin(); err != nil {
 		return result, errors.Trace(err)
 	}
 	for i, arg := range args.Args {
-		// TODO(wallyworld) - add backend specific config validation
-		err := s.state.AddSecretBackend(state.CreateSecretBackendParams{
-			Name:                arg.Name,
-			Backend:             arg.Backend,
-			TokenRotateInterval: arg.TokenRotateInterval,
-			Config:              arg.Config,
-		})
+		err := s.createBackend(arg)
 		result.Results[i].Error = apiservererrors.ServerError(err)
 	}
 	return result, nil
 }
 
+func (s *SecretBackendsAPI) createBackend(arg params.SecretBackend) error {
+	if arg.Name == "" {
+		return errors.NotValidf("missing backend name")
+	}
+	if arg.Name == juju.Backend || arg.Name == kubernetes.Backend {
```

Comment:
this check is to ensure that someone cannot create a backend called "juju" because that clashes with the fixed built in "juju" backend . It's just to help avoid a confusing situation. Same with "kubernetes".

Valid backend type is checked elsewhere